opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchScrollRequest.java

```diff
@@ -151,7 +151,40 @@ public void reset() {
    * @return a string representing the scroll request.
    */
   @Override
-  public String toCursor() {
-    return scrollId;
+  public boolean hasAnotherBatch() {
+    return !needClean && scrollId != null && !scrollId.equals("");
+  }
+
+  @Override
+  public void writeTo(StreamOutput out) throws IOException {
+    initialSearchRequest.writeTo(out);
+    out.writeTimeValue(scrollTimeout);
+    if (!needClean) {
+      // If needClean is true, there is no more data to get from OpenSearch and scrollId is
+      // used only to clean up OpenSearch context.
+
+      out.writeString(scrollId == null? "" : scrollId);
```

Comment:
I think you need to throw an exception (`NoCursorException`) if scroll is finished